src/runtime/virtcontainers/sandbox.go

```diff
@@ -2012,9 +2012,60 @@ func (s *Sandbox) updateResources(ctx context.Context) error {
 	}
 	s.Logger().Debugf("Sandbox CPUs: %d", newCPUs)
 
-	// Update Memory
-	s.Logger().WithField("memory-sandbox-size-byte", sandboxMemoryByte).Debugf("Request to hypervisor to update memory")
+	// Update Memory --
+	// If we're using ACPI hotplug for memory, there's a limitation on the amount of memory which can be hotplugged at a single time.
+	// We must have enough free memory in the guest kernel to cover 64bytes per (4KiB) page of memory added for mem_map.
+	// See https://github.com/kata-containers/kata-containers/issues/4847 for more details.
+	// For typical pod lifecycle, we expect that each container is added then we start the workloads. based on this, we'll "assume" that majority
+	// of the guest memory is readily available. From experimentation, we see that we can add approximately 48 times what is already provided to
+	// the guest workload. For example, a 256 MiB guest should be able to accomodate hotplugging 12 GiB  of memory.
```

Comment:
Can this be impacted by amount of memory currently in use within the guest. In our case we do the hot plug really early, so it should not be an issue. But if we hot plug later we may need to take that into consideration.

Reply:
Yes, for sure. If you later try to add a very large container, and the others are using near their limit, there's a good chance the hot plug will fail.  Given expected pod lifecycle, I don't expect this scenario.  This PR should address the existing common scenario, but we're using a heuristic here.  Ideally, when triggering the hot plug event we'd have a better signal to see if we were able to actually add the memory or not.  This is disjoint today.